packages/eslint-plugin/src/configs/typescript.ts

```diff
@@ -0,0 +1,24 @@
+import type { TSESLint } from "@typescript-eslint/utils"
+
+export = {
+  parserOptions: {
+    parser: "@typescript-eslint/parser",
+  },
+  plugins: ["@typescript-eslint", "tsdoc"],
+  extends: ["plugin:@typescript-eslint/recommended"],
+  rules: {
+    "@typescript-eslint/no-var-requires": ["off"],
+  },
+  overrides: [
+    {
+      files: ["*.ts"],
+      rules: {
+        "tsdoc/syntax": "error",
+        // This rule is disabled above to avoid forcing ESM syntax on regular JS files
```

Comment:
The disabled rule would raise on every usage of `require` in our regular JavaScript files and require them to be turned into `import` statements instead. Our tooling doesn't support that across the board, but our TypeScript files _can_ abide by that rule (because TypeScript supports that syntax out of the box). If we wanted to enable this rule everywhere we would need to change the node scripts we run directly (locales, tapes, etc) to `mjs`... or run them with `ts-node`.